### DISCUSSION TITLE
Using Axmol as a replacement engine in the last phase =/

### DISCUSSION THREAD

**Discussion by tirinkun** [axmolengine/axmol#1129] [General]
Hi, I've coded a private game with 3.17.2 Cocos2dx, it nearly done until I found the memory bug, it randomly crash upon changing scene (dealloc called)... something look like what described here : https://discuss.cocos2d-x.org/t/label-crash-in-the-wild-ios-cocos2d-x-v4/55250, and luckily I found this forum.

I would wanna ask if I could use Axmol as engine with my old code (for cocos2dx 3.17.2) with a few changes ( I guess it is still cocos2dx v4 ), ... it was mostly develop in 11.6.8 Mac OS (Big Sur) and XCode 13.2.1.

Thanks for your patient reading.

**Comment by rh101** [comment]
> I would wanna ask if I could use Axmol as engine with my old code (for cocos2dx 3.17.2) with a few changes ( I guess it is still cocos2dx v4 ),

Yes, and it should only take you a few hours at most to get it up and running on Axmol.

The quickest way would be to generate a new project using `axmol new ...` etc., then move your source code and resources over to that new project.  The reason is the `CMakeLists.txt` is significantly different from what is generated by Cocos2d-x, so it's best to start from one generated by Axmol, and copy over any custom code to it.  The same applies for any changes you need to make to the Android build files.

Other than that, if you run into any compilation errors with Axmol, then it should only take a search and replace to fix any issues.
